core/src/main/kotlin/plugability/DokkaContext.kt

```diff
@@ -67,14 +67,32 @@ private class DokkaContextConfigurationImpl(
         VISITED;
     }
 
-    internal val verticesWithState = mutableMapOf<Extension<*>, State>()
-    internal val adjacencyList: MutableMap<Extension<*>, MutableList<Extension<*>>> = mutableMapOf()
+    private sealed class Suppression {
+        data class ByExtension(val extension: Extension<*, *, *>) : Suppression() {
+            override fun toString() = extension.toString()
+        }
+
+        data class ByPlugin(val plugin: DokkaPlugin) : Suppression() {
+            override fun toString() = "Plugin ${plugin::class.qualifiedName}"
+        }
+    }
 
-    private fun topologicalSort() {
+    private val rawExtensions = mutableListOf<Extension<*, *, *>>()
+    private val rawAdjacencyList = mutableMapOf<Extension<*, *, *>, MutableList<Extension<*, *, *>>>()
+    private val suppressedExtensions = mutableMapOf<Extension<*, *, *>, MutableList<Suppression>>()
 
-        val result: MutableList<Extension<*>> = mutableListOf()
+    fun topologicallySortAndPrune() {
```

Comment:
I think we could make it `internal`

Reply:
This function is defined in private class so there is no way to call it from outside of the module.